src/state.rs

```diff
@@ -426,7 +426,7 @@ impl AtomicPosition {
         let (new, remainder) = ((diff / INTERVAL), (diff % INTERVAL));
         // We add `new` to `capacity`, subtract one for returning `true` from here,
         // then make sure it does not exceed a maximum of `MAX_BURST`.
-        capacity = Ord::min(MAX_BURST, capacity + new as u8 - 1);
+        capacity = Ord::min(MAX_BURST, u64::from(capacity).saturating_add(new) as u8 - 1);
```

Comment:
I think we do the saturating add in the `u8` domain, not in the `u64` domain.

Reply:
@djc is there a saturating constructor for `u8` that sets the value to `u8::MAX` if it wouldn't fit? I don't think we want to write `new as u8` here, since if `new` is particularly large, the most-significant bits will be cut off, and you could instead end up with a `new` value which is very small. That would cause the `capacity` to be assigned a lower than expected value. That being said, I don't understand the performance implications of this code (there don't seem to be any benches?), so I don't know which kinds of operations are expensive.

Reply:
I've used the fastbar example as a bench. I don't think integer operations will make a substantial impact, though, so we can just do what's necessary here, something like `Ord::min(capacity, u8::MAX as u128) as u8` or something?

Reply:
Also as long as you're fixing things here, please also audit the very similar logic in the `draw_target::RateLimiter`.

Reply:
The code seems to end up ugly no matter what configuration I try because all of the numeric types are different. The latest version upcasts, saturates, and then downcasts. In particular, I'm trying to avoid the use of `as`, since it could lead to weird bugs due to the fact that it truncates rather than saturates.